plugin.go

```diff
@@ -225,11 +234,78 @@ func (s *Plugin) Middleware(next http.Handler) http.Handler { //nolint:gocognit,
 			}
 		}
 
+		if s.gzip && r.Header.Get("Accept-Encoding") == "gzip" {
+			s.log.Debug("gzip compression requested")
+			cf, err := s.compressContent(fp, f)
+			if err != nil {
+				s.log.Error("failed to compress content", zap.Error(err))
+				http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+				return
+			}
+			err = f.Close()
+			if err != nil {
+				s.log.Error("failed to close file", zap.Error(err))
+			}
+			f = cf
```

Comment:
I'm not sure what is that. You're closing a file and then reassign another file to the closed one to be closed by the initial defer?